tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs

```diff
@@ -50,6 +52,77 @@ public async Task VerifyRedisResource()
         Assert.Equal("value", value);
     }
 
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyWithRedisInsight()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+
+        using var builder = TestDistributedApplicationBuilder.CreateWithTestContainerRegistry(testOutputHelper);
+
+        var redis1 = builder.AddRedis("redis-1");
+        var redis2 = builder.AddRedis("redis-2").WithRedisInsight();
+
+        builder.Services.AddHttpClient();
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var factory = app.Services.GetRequiredService<IHttpClientFactory>();
+        var client = factory.CreateClient();
+
+        var redisInsightResource = builder.Resources.OfType<RedisInsightResource>().SingleOrDefault();
+
+        var insightEndpoint = redisInsightResource!.PrimaryEndpoint;
```

Comment:
If we are sure that `redisInsightResource` is not going to be null, then we could instead use `.Single()` in the line above and that throw if this is not true.